netlink.go

```diff
@@ -109,21 +147,62 @@ func (c *NetlinkClient) RemoveRoute(vni VNI, dest Destination, hop NextHop) erro
 		return fmt.Errorf("cannot parse destination prefix: %v", err)
 	}
 
-	encap := netlink.IP6tnlEncap{
-		Dst: net.ParseIP(hop.TargetAddress.String()),
-		Src: net.ParseIP("::"), // what source ip to put here? Metalbond object, m, does not contain this info yet.
-	}
-
 	route := &netlink.Route{
-		LinkIndex: c.tunDevice.Attrs().Index,
-		Dst:       dst,
-		Encap:     &encap,
-		Table:     table,
-		Protocol:  METALBOND_RT_PROTO,
+		Dst:      dst,
+		Table:    table,
+		Protocol: METALBOND_RT_PROTO,
 	} // by default, the route is already installed into the kernel table without explicite specification
 
-	if err := netlink.RouteDel(route); err != nil {
-		return fmt.Errorf("cannot remove route to %s (table %d) from kernel: %v", dest, table, err)
+	err, _ = c.routeTable.RemoveNextHop(vni, dest, hop, c.mbp)
+	if err != nil {
+		return fmt.Errorf("cannot add route to internal table vni: %s dest: %s hop: %s error: %v", vni, dest, hop, err)
+	}
+
+	if c.config.PreferNetwork != nil {
```

Comment:
This code looks very similar to the one above. Can we move this into a helper method?